mern-ecommerce/controllers/product.js

```diff
@@ -8,7 +8,7 @@ exports.productById = (req, res, next, id) => {
   Product.findById(id)
     .populate('category')
     .exec((err, product) => {
-      if (err || !product) {
+      if (!err || !product) {
```

Comment:
The condition `if (!err || !product)` is incorrect. It should be `if (err || !product)`. The current condition will evaluate to true if either `err` is falsy or `product` is falsy, which is not the intended behavior.